packaging/grafana-agent-flow/windows/install_script.nsis

```diff
@@ -105,12 +116,24 @@ Function InitializeRegistry
     Pop $0 # Ignore return result
   ${EndIf}
 
+  ${If} $Config != ""
+    StrCpy $ConfigFlag "$Config"
+  ${Else}
+    StrCpy $ConfigFlag "$INSTDIR\config.river"
+  ${EndIf}
+
+  ${If} $DisableReporting == "yes"
+    StrCpy $DisableReportingFlag "--disable-reporting\0"
+  ${Else}
+    StrCpy $DisableReportingFlag ""
+  ${EndIf}
+
   # Define the arguments key, which holds arguments to pass to the
   # service.
   nsExec::ExecToLog 'Reg.exe query "${REGKEY}" /reg:64 /v Arguments'
   Pop $0
   ${If} $0 == 1
-    nsExec::ExecToLog 'Reg.exe add "${REGKEY}" /reg:64 /v Arguments /t REG_MULTI_SZ /d "run"\0"$INSTDIR\config.river"\0"--storage.path=$APPDATA\${APPNAME}\data"\0'
+    nsExec::ExecToLog 'Reg.exe add "${REGKEY}" /reg:64 /v Arguments /t REG_MULTI_SZ /d "run"\0"$ConfigFlag"\0"--storage.path=$APPDATA\${APPNAME}\data"\0"$DisableReportingFlag"'
```

Comment:
For some reason the quotes look off, though I assume you have tested the scenarios?